components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/AuthenticationRequestBuilder.java

```diff
@@ -0,0 +1,181 @@
+        return eventBuilder.build();
+    }
+
+    private User getUserForEventBuilder(AuthenticatedUser authenticatedUser)
+            throws ActionExecutionRequestBuilderException {
+
+        try {
+            return new AuthenticatingUser(authenticatedUser.getUserId(), authenticatedUser);
+        } catch (UserIdNotFoundException e) {
+            throw new ActionExecutionRequestBuilderException("User ID not found for current authenticated user.", e);
+        }
+    }
+
+    private Organization getOrganizationForEventBuilder(AuthenticatedUser authenticatedUser)
+            throws ActionExecutionRequestBuilderException {
+
+        String organizationId = authenticatedUser.getUserResidentOrganization();
+        try {
+            if (organizationId != null && !organizationId.isEmpty()) {
+                String organizationName = AuthenticatorAdapterDataHolder.getInstance().getOrganizationManager()
+                        .getOrganizationNameById(authenticatedUser.getUserResidentOrganization());
+                return new Organization(authenticatedUser.getUserResidentOrganization(), organizationName);
+            }
+        } catch (OrganizationManagementException e) {
+            LOG.debug("Error occurred while retrieving organization name of the authorized user: " + organizationId, e);
+            return null;
+        }
+
+        LOG.debug("The organization id is not set for the authorized user with username: "
+                + authenticatedUser.getUserName());
+        return null;
+    }
+
+    private AuthenticatedStep[] getAuthenticatedStepsForEventBuilder(AuthenticationContext context) {
+
+        ArrayList<AuthenticatedStep> authenticatedSteps = new ArrayList<>();
+        int stepIndex = 1;
+        for (AuthHistory step: context.getAuthenticationStepHistory()) {
+            authenticatedSteps.add(new AuthenticatedStep(stepIndex, step));
+            stepIndex += 1;
+        }
+
+        return authenticatedSteps.toArray(new AuthenticatedStep[0]);
+    }
+
+    private Request getRequest(HttpServletRequest request) {
+
+        List<Header> additionalHeaders = new ArrayList<>();
+        List<Param> additionalParams = new ArrayList<>();
+
+        Enumeration<String> headers = request.getHeaderNames();
+        while (headers.hasMoreElements()) {
+            String headerElement = headers.nextElement();
+            Header header = new Header(headerElement, new String[]{request.getHeader(headerElement)});
+            additionalHeaders.add(header);
+        }
+
+        Enumeration<String> requestParameters = request.getParameterNames();
+        while (requestParameters.hasMoreElements()) {
+            String paramElement = requestParameters.nextElement();
+            Param param = new Param(paramElement, new String[]{request.getParameter(paramElement)});
+            additionalParams.add(param);
+        }
+
+        return new AuthenticationRequest(additionalHeaders, additionalParams);
+    }
```

Comment:
Let's not add headers or parameters for the moment. Since these may include sensitive info we need to provide a facility to let them configure what headers are params should be included.
Also when it comes to asgardeo it may send internal headers and parameters that should not be passed to the extension